client/ayon_maya/api/lib.py

```diff
@@ -4544,3 +4557,85 @@ def get_scene_units_settings(project_settings=None)-> tuple[str, str]:
     linear_unit = scene_units.get("linear_units", "cm")
     angular_unit = scene_units.get("angular_units", "deg")
     return linear_unit, angular_unit
+
+
+def connect_texture_reference_objects(
+        texture_connections, nodes_by_id, tx_ref_nodes_by_id):
+    """Connect texture reference object nodes to the target object
+    nodes if there is one.
+
+    Args:
+        texture_connections (list): list of texture reference
+        objects connection data
+        nodes_by_id (dict): The dict with node ids.
+        tx_ref_nodes_by_id (dict): The dict with texture reference node ids.
+    """
```

Comment:
None of the logic here is _really_ specific to texture reference objects - it's just processing the connections - similar to [here](https://github.com/ynput/ayon-core/blob/b0ac87b7b12006dc8b3fffe441b91a884cc65ce5/client/ayon_core/tools/workfiles/models/workfiles.py#L360-L366).

I'd name this `apply_connections(connections, source_nodes_by_id, target_nodes_by_id)`